MBBSEmu/HostProcess/HostRoutines/MenuRoutines.cs

```diff
@@ -28,22 +27,25 @@ public class MenuRoutines : IHostRoutine
         private readonly IAccountKeyRepository _accountKeyRepository;
         private readonly AppSettings _configuration;
         private readonly IGlobalCache _globalCache;
+        private readonly PointerDictionary<SessionBase> _channelDictionary;
 
-        public MenuRoutines(IResourceManager resourceManager, IAccountRepository accountRepository, AppSettings configuration, IGlobalCache globalCache, IAccountKeyRepository accountKeyRepository)
+        public MenuRoutines(IResourceManager resourceManager, IAccountRepository accountRepository, AppSettings configuration, IGlobalCache globalCache, IAccountKeyRepository accountKeyRepository, PointerDictionary<SessionBase> channelDictionary)
         {
             _resourceManager = resourceManager;
             _accountRepository = accountRepository;
             _accountKeyRepository = accountKeyRepository;
             _configuration = configuration;
             _globalCache = globalCache;
+            _channelDictionary = channelDictionary;
         }
 
         /// <summary>
         ///     Invokes Method based on the Session State
         /// </summary>
         /// <param name="session"></param>
         /// <param name="modules"></param>
-        public bool ProcessSessionState(SessionBase session, Dictionary<string, MbbsModule> modules)
+        /// <param name="sessions"></param>
+        public bool ProcessSessionState(SessionBase session, Dictionary<string, MbbsModule> modules, PointerDictionary<SessionBase> channelDictionary)
```

Comment:
We can use the instance `PointerDictionary<SessionBase>` passed in on the constructor, don't need to pass it here as they should both be a reference to the same object.